### PR TITLE
CEDS 2640 Add Ducr Part journey

### PR DESCRIPTION
**Pull request by PatrykRudnicki**
- CEDS-2640 Add Ducr Part journey
- CEDS-2640 Add feature switch to service and DUCR Part feature

### CODE REVIEW

File: test/unit/models/UcrBlockSpec.scala [hunk 1]
```diff
@@ -31,18 +31,18 @@ class UcrBlockSpec extends WordSpec with MustMatchers {
         val ucrBlock = UcrBlock(ucr = validDucr, ucrType = Ducr.codeValue)
```
Comment (SteveSugden) [inline test/unit/models/UcrBlockSpec.scala:31]: Why have you changed this test.  UcrBlock hasn't changed

Comment (PatrykRudnicki) [reply test/unit/models/UcrBlockSpec.scala]: Good catch, it was when I tried to remove `ucrPartNo` I renamed fullUcr to ucr and didn't change it back. Thanks

Comment (PatrykRudnicki) [reply test/unit/models/UcrBlockSpec.scala]: but this confirm what we discussed yesterday, that ucr field contains the ducr with part

Comment (PatrykRudnicki) [reply test/unit/models/UcrBlockSpec.scala]: I pushed a fix in separate commit for it

File: app/models/cache/Answers.scala [hunk 2]
```diff
@@ -66,6 +66,8 @@ object ArrivalAnswers {
       ArrivalAnswers(consignmentReferences = Some(ConsignmentReferences(ConsignmentReferenceType.M, ucrBlock.ucr)))
     case Some(ucrBlock) if ucrBlock.ucrType.equals(Ducr.codeValue) =>
       ArrivalAnswers(consignmentReferences = Some(ConsignmentReferences(ConsignmentReferenceType.D, ucrBlock.ucr)))
+    case Some(ucrBlock) if ucrBlock.ucrType.equals(DucrPart.codeValue) =>
+      ArrivalAnswers(consignmentReferences = Some(ConsignmentReferences(ConsignmentReferenceType.DP, ucrBlock.ucr)))
```
Comment (SteveSugden) [inline app/models/cache/Answers.scala:70]: Would all of these DUCR Part answers be safer if they were using the ucrBlock.fullUcr method?    I know they currently return the same thing but if we decide to change the UcrBlock implementation to store the partId separately then this code here won't need to change.

Applies to all changes to this file

File: app/models/viewmodels/notificationspage/NotificationPageSingleElementFactory.scala [hunk 3]
```diff
@@ -34,14 +34,16 @@ class NotificationPageSingleElementFactory @Inject()(responseConverterProvider:
 
   def build(submission: Submission)(implicit messages: Messages): NotificationsPageSingleElement =
     submission.actionType match {
-      case Arrival | RetrospectiveArrival | Departure | DucrDisassociation | MucrAssociation | MucrDisassociation | ShutMucr =>
+      case Arrival | RetrospectiveArrival | Departure | DucrDisassociation | DucrPartDisassociation | MucrAssociation | MucrDisassociation |
+          ShutMucr =>
         buildForRequest(submission)
-      case DucrAssociation => buildForDucrAssociation(submission)
+      case DucrAssociation     => buildForDucrAssociation(submission)
```
Comment (SteveSugden) [inline app/models/viewmodels/notificationspage/NotificationPageSingleElementFactory.scala:40]: Why is this different to ext movements?  There both DucrAssociation and DucrPartAssociation -> buildForDucrAssociation()

File: app/models/cache/Answers.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/models/cache/Answers.scala]: Yeah, pushed added this to review fixes commit. Good catch

File: app/models/viewmodels/notificationspage/NotificationPageSingleElementFactory.scala (hunk 3 above)
Comment (PatrykRudnicki) [reply app/models/viewmodels/notificationspage/NotificationPageSingleElementFactory.scala:40]: This is the bug in the External Movements service, I'll create a fix for it there after I merge internal

Comment (SteveSugden) [reply app/models/viewmodels/notificationspage/NotificationPageSingleElementFactory.scala:40]: OK.  Great!

**Pull request merged**